lib/FusionInventory/Agent/Task/Inventory/Generic/Ipmi/Fru.pm

```diff
@@ -0,0 +1,55 @@
+#
+# OcsInventory agent - IPMI lan channel report
+#
+# Copyright (c) 2008 Jean Parpaillon <jean.parpaillon@kerlabs.com>
+#
+# The Intelligent Platform Management Interface (IPMI) specification
+# defines a set of common interfaces to a computer system which system
+# administrators can use to monitor system health and manage the
+# system. The IPMI consists of a main controller called the Baseboard
+# Management Controller (BMC) and other satellite controllers.
+#
+package FusionInventory::Agent::Task::Inventory::Generic::Ipmi::Fru;
+
+use strict;
+use warnings;
+
+use FusionInventory::Agent::Tools;
+use FusionInventory::Agent::Tools::Generic;
+
+sub isEnabled {
+    my (%params) = @_;
+    return 0 if $params{no_category}->{psu};
+    return 1;
+}
+
+sub doInventory {
+    my (%params) = @_;
+
+    my $inventory = $params{inventory};
+    my $logger    = $params{logger};
+
+    my $fru = getIpmiFru(logger => $logger);
+    my $psu;
+
+    foreach my $descr (keys %$fru) {
+        next unless $descr =~ /^PS(\d+)/;
+
+        $psu = {
+            PARTNUM => $fru->{$descr}->{data}->{'Board Part Number'},
+            SERIAL  => $fru->{$descr}->{data}->{'Board Serial'},
+            POWER   => $fru->{$descr}->{data}->{'Max Power Capacity'},
+            VENDOR  => $fru->{$descr}->{data}->{'Board Mfg'},
```

Comment:
Here I saw, there's also a "Product Manufacturer" available.

Reply:
This is what i have:

```
FRU Device Description : PS1 (ID 1)
 Board Mfg Date        : Tue Sep  3 19:44:00 2013
 Board Mfg             : DELL
 Board Product         : PWR SPLY,750WP,RDNT,DELTA     
 Board Serial          : CN1797238QA8EI
 Board Part Number     : 05NF18A01

FRU Device Description : PS2 (ID 2)
 Board Mfg Date        : Tue Sep  3 19:23:00 2013
 Board Mfg             : DELL
 Board Product         : PWR SPLY,750WP,RDNT,DELTA     
 Board Serial          : CN1797238QA8B4
 Board Part Number     : 05NF18A01
```

Reply:
Hi @po1vo 
can you also provide the `dmidecode --type 39` output from the same machine ? So I'll integrate it in unittests. Thank you.

Reply:
@g-bougard , it's blank. We have Dells and HPs, and either dmidecode or ipmitool have PSU data, not both at the same.

What about my questions? The work has stalled until i have answers. :)

Reply:
g-bougard, I have it, I prepare files and send you by mail